qiskit_nature/second_q/algorithms/initial_points/mp2_initial_point.py

```diff
@@ -14,75 +14,87 @@
 
 from __future__ import annotations
 
-from dataclasses import dataclass
-
 import numpy as np
-from qiskit_nature.exceptions import QiskitNatureError
 
+from qiskit_nature.exceptions import QiskitNatureError
 from qiskit_nature.second_q.circuit.library import UCC
-from qiskit_nature.second_q.properties import ElectronicEnergy
+from qiskit_nature.second_q.properties import ElectronicEnergy, GroupedSecondQuantizedProperty
 from qiskit_nature.second_q.properties.bases import ElectronicBasis
-from qiskit_nature.second_q.properties.integrals.electronic_integrals import (
-    ElectronicIntegrals,
-)
-from qiskit_nature.second_q.properties.second_quantized_property import (
-    GroupedSecondQuantizedProperty,
-)
+from qiskit_nature.second_q.properties.integrals import ElectronicIntegrals
+
 
 from .initial_point import InitialPoint
 
 
-@dataclass(frozen=True)
-class _Correction:
-    """Class for storing data about each MP2 correction term."""
+def _compute_mp2(
+    integral_matrix: np.ndarray, orbital_energies: np.ndarray
+) -> tuple[np.ndarray, float]:
+    """Compute the T2 amplitudes and MP2 energy correction.
+
+    Args:
+        integral_matrix: The two-body molecular orbitals matrix.
+        orbital_energies: The orbital energies.
+
+    Returns:
+        T amplitudes t2[i, j, a, b] (i, j in occupied, a, b in virtual).
+        The MP2 energy correction.
+
+    """
+    num_occ = len(orbital_energies[orbital_energies < 0.0])
```

Comment:
We pass in the full `grouped_property`, so we can access it. It'll just need some extra checks.

Reply:
Out of curiosity, what sort of molecules would feature this?

Reply:
When you start looking at very complex metalorganics, depending on the quality of your initial method, I've seen this happen.
It's rare, but I'd rather not rely on it, if we have a better way :+1:

Reply:
I opted to use `occupied_alpha` as this has a predictable format. Am I right in thinking that this will always be present in a problem, or should I check for it as we do for the ElectronicEnergy etc?